### PR TITLE
map matcher: removing exceptions

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: updated tests
Docs Changes: n/a
Release Notes: n/a
https://github.com/envoyproxy/envoy-mobile/issues/176

### CODE REVIEW

File: source/common/matcher/exact_map_matcher.h [hunk 1]
```diff
@@ -11,16 +11,27 @@ namespace Matcher {
...
 template <class DataType> class ExactMapMatcher : public MapMatcher<DataType> {
 public:
-  ExactMapMatcher(DataInputPtr<DataType>&& data_input,
-                  absl::optional<OnMatch<DataType>> on_no_match)
-      : MapMatcher<DataType>(std::move(data_input), std::move(on_no_match)) {}
+  static absl::StatusOr<std::unique_ptr<ExactMapMatcher>>
+  create(DataInputPtr<DataType>&& data_input, absl::optional<OnMatch<DataType>> on_no_match) {
+    absl::Status creation_status = absl::OkStatus();
+    auto ret = std::unique_ptr<ExactMapMatcher<DataType>>(
+        new ExactMapMatcher<DataType>(std::move(data_input), on_no_match, creation_status));
```
Comment (tyxia) [inline source/common/matcher/exact_map_matcher.h:18]: nit: Can we do `make_unique` here directly?
https://abseil.io/tips/126

**Review by tyxia — APPROVED** [review]
LGTM. Thanks!

File: source/common/matcher/exact_map_matcher.h (hunk 1 above)
Comment (alyssawilk) [reply source/common/matcher/exact_map_matcher.h:18]: I think not unless the constructor is public, which Yan prefered not to do when we have a status-fail in the constructor

**Review by wbpcode — APPROVED** [review]

**Pull request merged**